forge-game/src/main/java/forge/game/trigger/TriggerHandler.java

```diff
@@ -564,6 +564,15 @@ private void runSingleTriggerInternal(final Trigger regtrig, final Map<AbilityKe
             Card triggeredCard = ((Card) sa.getTriggeringObject(AbilityKey.Card));
             host.addRemembered(triggeredCard);
         }
+        if (regtrig.hasParam("LifePaid")) {
+            final SpellAbility trigSA = (SpellAbility) runParams.get(AbilityKey.CastSA);
+            int amount = trigSA.getAmountLifePaid();
+            if (amount > 0) {
+                host.addRemembered(amount);
```

Comment:
Remembering it on the host would probably break if you have such ability on the stack twice, one with life paid and one without?

Reply:
Good call, different amounts might also get messy. Hmm, back to the drawing board – any ideas?

Reply:
Imo the CastSA should persist that info, so you should query TriggeringObjects directly

Reply:
Good instinct on this – I did test it quickly with the old code and it definitely didn't handle multiple triggers well. This new version seems to handle this case correctly